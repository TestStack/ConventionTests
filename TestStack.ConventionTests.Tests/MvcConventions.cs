namespace TestStack.ConventionTests.Tests
{
    using ApprovalTests;
    using ApprovalTests.Reporters;
    using NUnit.Framework;
    using TestAssembly.Controllers;
    using TestStack.ConventionTests.ConventionData;
    using TestStack.ConventionTests.Conventions;

    [TestFixture]
    [UseReporter(typeof(DiffReporter))] 
    public class MvcConventions
    {
        [Test]
        public void controller_conventions()
        {
            var types = Types.InAssemblyOf<TestController>();
            var convention = new MvcControllerNameAndBaseClassConvention();

            var ex = Assert.Throws<ConventionFailedException>(() => Convention.Is(convention, types));
            Approvals.Verify(ex.Message);
        }

        [Test]
        public void api_controller_conventions()
        {
            var types = Types.InAssemblyOf<TestController>();
            var convention = new ApiControllerNamingAndBaseClassConvention();

            var ex = Assert.Throws<ConventionFailedException>(() => Convention.Is(convention, types));
            Approvals.Verify(ex.Message);
        } 
    }
}