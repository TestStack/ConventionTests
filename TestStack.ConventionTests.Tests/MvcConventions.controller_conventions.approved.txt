'Mvc Controllers must be suffixed with Controller' for 'Types in TestAssembly'
------------------------------------------------------------------------------

	TestAssembly.Controllers.FooControler

'Types named *Controller must inherit from ApiController or Controller' for 'Types in TestAssembly'
---------------------------------------------------------------------------------------------------

	TestAssembly.Controllers.TestApiController
	TestAssembly.Controllers.TestController
